src/inc/corinfo.h

```diff
@@ -832,7 +832,7 @@ enum CorInfoFlag
     CORINFO_FLG_INTRINSIC             = 0x00400000, // This method MAY have an intrinsic ID
     CORINFO_FLG_CONSTRUCTOR           = 0x00800000, // This method is an instance or type initializer
     CORINFO_FLG_AGGRESSIVE_OPT        = 0x01000000, // The method may contain hot code and should be aggressively optimized if possible
-//  CORINFO_FLG_UNUSED                = 0x02000000,
+    CORINFO_FLG_ALLOW_TIER0_TO_TIER1  = 0x02000000, // Indicates that for a tier 0 compilation request, the JIT may choose to switch to tier 1 if appropriate
```

Comment:
I think of this slightly differently - that the runtime made a request for an indeterminate optimization level and the JIT resolved it to a specific optimization level based on some criteria. Although arguably this is just semantics, my concern about describing it this way is that new developers are likely to take flags/constants that say Tier0 at face value without necessarily noticing that there is a special case that converts 0 into 1. Using alternate terminology such as TIER0_OR_1 or UNRESOLVED_TIER (and not specifying TIER0) feels much harder to misunderstand.

Reply:
The requested intent from the VM seems to be "JIT quickly". The JIT may choose to do so or not for various reasons. For example if the method only has a small loop and a large amount of code outside it may choose to optimize the loop and to not optimize code outside. In that case the VM would still have to consider that as unoptimized code. Or if the JIT finds that the tier 0 code would be too slow because of too many high-impact optimizations being missed, it may choose to generate fully optimized code, in which case the VM would consider that as optimized code. Maybe something like "PREFER_TIER0" would be clearer intent, or something more generic like "JIT_QUICKLY".

I think I'll leave the JIT config flags as they are for now. For this and the preview 4 PR I would like to start with some decent names for the config options. It may not be a big deal though if we have to rename them again.